lib/pushmi_pullyu/cli.rb

```diff
@@ -181,14 +182,20 @@ def rotate_logs
   end
 
   def run_preservation_cycle
-    item = queue.wait_next_item
-    return unless item.present?
+    entity_json = JSON.parse(queue.wait_next_item)
+    entity = {
+      type: entity_json['type'],
+      uuid: entity_json['uuid']
+    }
+    return unless entity.present?
```

Comment:
You need to be careful here! This will always be true! Your asserting if a hash with key value pairs is true which it is always true. For example lets say `entity_json['type']` and `entity_json['uuid']` are both nil...
```
irb(main):001:0> entity = { type: nil, uuid: nil }
=> {:type=>nil, :uuid=>nil}
irb(main):002:0> entity.present?
=> true
```

Reply:
Right! Will make the change.